cloud-nio/cloud-nio-impl-drs/src/main/scala/cloud/nio/impl/drs/DrsCloudNioFileProvider.scala

```diff
@@ -0,0 +1,61 @@
+package cloud.nio.impl.drs
+
+import java.nio.channels.{ReadableByteChannel, WritableByteChannel}
+
+import cloud.nio.spi.{CloudNioFileList, CloudNioFileProvider, CloudNioRegularFileAttributes}
+import com.typesafe.config.Config
+import org.apache.http.HttpStatus
+import org.apache.http.client.methods.CloseableHttpResponse
+import org.apache.http.impl.client.{CloseableHttpClient, HttpClientBuilder}
+
+import scala.util.Try
+
+
+class DrsCloudNioFileProvider(config: Config, scheme: String, drsPathResolver: DrsPathResolver) extends CloudNioFileProvider {
+
+  private def getDrsPath(cloudHost: String, cloudPath: String): String = s"$scheme://$cloudHost/$cloudPath"
+
+  private def checkIfPathExistsThroughMartha(drsPath: String): Boolean = {
+    val httpClient: CloseableHttpClient = HttpClientBuilder.create().build()
```

Comment:
ToL: Can the builder be pulled outside this function or do we need to create a new one every time ?

Reply:
Surprisingly it does work! Thanks :)